로그인 화면에 온보딩 화면 적용

junu0516 opened the pull request:
### 🔖 관련 이슈
- #34 

<br> 

### ⚒️ 작업 내역

- [x] 커스텀 LaunchScreen 구현
- [x] 배너 이미지 배치를 위한 컬렉션뷰 LoginViewControleller에 적용
- [x] 닉네임 설정을 위한 ViewType, ViewModel 추가 구현

<br>

### 💻 리뷰어 가이드
> 리뷰어가 작업을 확인할 수 있는 방법, 기대되는 정상적인 동작 내용을 작성

- 앱 기동 후 LaunchScreen 노출 > 로그인 화면 출력 > 카카오/애플 로그인 버튼 터치 > 닉네임 화면 출력 순서로 확인하면 됩니다.
- 닉네임 입력 시 2~5글자 이내일 때, 필수 확인 요소 선택했을 때만 다음 버튼이 활성화되면 됩니다.

<br>

### 📝 부가 설명

- 일부 화면 크기가 작은 기기에 대해서 별도로 레이아웃을 추후 적용하려 합니다.
- NickNameViewModel의 확인 요소 선택 검증에 대한 로직이 좀 복잡하게 작성되서 이 부분도 추후 다시 간소화하려 합니다.
- 더 이상 작업하면 해당 PR에 담기는 코드량이 너무 많아질듯해서 우선 화면 구현 정도로 끝내고 추가 기능 구현 및 기존 기능 개선은 별도의 이슈를 파서 진행하려 합니다.

<br> 

### 📑 첨부 자료
> 작업 관련 참고 자료, 스크린샷, 동영상 등을 첨부

- 내용 작성 (Before/After 비교 방식으로 스크린샷이나 동영상을 첨부하면 좋음)

|작업 이전|작업 이후|
|-----|-----|
|![Simulator Screen Recording - iPhone 12 mini - 2023-05-21 at 12 48 17](https://github.com/LifePoop/LifePoop_iOS/assets/68586291/34211ebc-62bc-481b-b60e-a6f8a39d93a8)|![Simulator Screen Recording - iPhone 12 mini - 2023-05-21 at 12 45 15](https://github.com/LifePoop/LifePoop_iOS/assets/68586291/67e80f47-1dab-4ba2-8890-9bbc6e82e32f)|

junu0516 committed:
Merge branch 'develop' into 34-onboardingScene

junu0516 committed:
#34 feat: LoginCoordinatorAction 케이스 추가

junu0516 committed:
#34 feat: 로그인 화면에 매너 이미지 수평 스크롤 형태로 배치하는 부분 적용, 로그인 버튼 배치 수정

- 추후 iPhone SE 등 모델에 대한 별도 레이아웃 처리 필요
- 배너 이미지 개수 및 배치에 대해 클라이언트쪽에서 정적으로 확정할 지, 서버에서 동적으로 결정할 지 확인 필요

junu0516 committed:
#34 feat: 가입 조건 선택을 위한 모델 객체, 컬렉션뷰 셀 형태 구현

junu0516 committed:
#34 feat: 닉네임 설정을 위한 NicknameViewController, NickNameViewModel 구현

junu0516 committed:
#34 feat: 커스텀 LaunchScreen 임시로 구현

- 화면 흐려지는 애니메이션 적용 필요
- 약 몇 초간 LaunchScreen을 띄울 지, 혹은 특정 동작을 수행하는 동안만 LaunchScreen을 띄울 지 확정 필요

junu0516 committed:
#34 remove: 불필요한 Rx+ConditionalTextField 파일 제거

junu0516 committed:
#34 feat: 수정된 부분에 맞춰 DefaultLoginCoordinator 내부의 화면 전환 흐름 구현 및 수정

sanghyeok-kim commented:
말씀하신대로 런치스크린 이후에 온보딩 화면이 나타나는게 뚝 끊겨보이긴하네요 ㅎㅎ 이건 나중에 수정해보죠

이외에 지금 바로 개선하면 좋을 것 같은 부분은
1. 닉네임 설정 화면이 push되면 textField를 firstResponder로 설정해주기
2. 4가지 옵션이 모두 선택되면, 전체 동의도 자동으로 선택되게 하기
3. 4가지 옵션이 모두 선택된 상태(따라서 전체 동의도 선택된 상태)에서, 하나라도 선택 해제되면, 전체 동의도 자동으로 선택 해제되게 하기

1번은 쉽게 수정할 수 있을 것 같은데
2번, 3번은 꼭 필요한건 아니라서 시간 나면 해주시고 아니면 넘어가셔도 됩니다!

추가적으로 수정하실 내용 있으시면 마저 커밋하시고 머지해주세요~

junu0516 commented:
오 1~4 다 생각 못했네요 ㅋㅋ 바로 적용할 수 있을 것 같아요. 여기에 추가 커밋해서 머지할께요~!

junu0516 committed:
#34 feat: ConditionalTextField 키보드 출력 로직 개선

- 키보드 Return 버튼 터치하면 내려가도록 처리
- 닉네임 화면에서 ConditionalTextField 이외의 영역 터치하면 키보드 내려가도록 처리
- 닉네임 화면 처음 push되면 키보드 자동으로 띄워지도록 처리

junu0516 committed:
#34 refactor: NicknameViewModel 안의 SwiftLint 걸리는 부분 수정

junu0516 committed:
#34 refactor: 바인딩 속성명 수정, 추후 리팩토링 해야할 부분 우선 주석 추가

junu0516 commented:
1,4는 몇 가지 추가해서 키보드 띄우는 로직 개선했고,
2,3은 본문에 적은 NicknameViewModel 안에서 조건 선택하는 로직 개선하면서 같이 해야할듯해서 우선 여기까지 하고 머지할께요!
